gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GCSManualClientImpl.java

```diff
@@ -0,0 +1,222 @@
+      throws IOException {
+    return gcsClientDelegate.listObjectInfo(bucketName, objectNamePrefix, listOptions);
+  }
+
+  @Override
+  public ListPage<GoogleCloudStorageItemInfo> listObjectInfoPage(
+      String bucketName, String objectNamePrefix, ListObjectOptions listOptions, String pageToken)
+      throws IOException {
+    return gcsClientDelegate.listObjectInfoPage(
+        bucketName, objectNamePrefix, listOptions, pageToken);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo getItemInfo(StorageResourceId resourceId) throws IOException {
+    return gcsClientDelegate.getItemInfo(resourceId);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> getItemInfos(List<StorageResourceId> resourceIds)
+      throws IOException {
+    return gcsClientDelegate.getItemInfos(resourceIds);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> updateItems(List<UpdatableItemInfo> itemInfoList)
+      throws IOException {
+    return gcsClientDelegate.updateItems(itemInfoList);
+  }
+
+  @Override
+  public void compose(
+      String bucketName, List<String> sources, String destination, String contentType)
+      throws IOException {
+    gcsClientDelegate.compose(bucketName, sources, destination, contentType);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo composeObjects(
+      List<StorageResourceId> sources, StorageResourceId destination, CreateObjectOptions options)
+      throws IOException {
+    return gcsClientDelegate.composeObjects(sources, destination, options);
+  }
+
+  @Override
+  public Map<String, Long> getStatistics() {
+    return gcsClientDelegate.getStatistics();
+  }
+
+  @Override
+  public void close() {
+    gcsClientDelegate.close();
+  }
+
+  public static class GCSManualClientImplBuilder {
+
+    private Credentials credentials;
+    private com.google.api.services.storage.Storage storage;
+    private HttpRequestInitializer httpRequestInitializer;
+    private GoogleCloudStorageOptions storageOptions;
+    private Function<List<AccessBoundary>, String> downscopedAccessTokenFn;
+
+    public GCSManualClientImplBuilder(
+        GoogleCloudStorageOptions storageOptions,
+        Credentials credentials,
+        Function<List<AccessBoundary>, String> downscopedAccessTokenFn) {
+      this.storageOptions = storageOptions;
+      this.credentials = credentials;
+      this.downscopedAccessTokenFn = downscopedAccessTokenFn;
+    }
+
+    @VisibleForTesting
+    public GCSManualClientImplBuilder withApairyClientStorage(
+        com.google.api.services.storage.Storage storage) {
+      this.storage = storage;
+      return this;
+    }
+
+    @VisibleForTesting
+    public GCSManualClientImplBuilder withHttpRequestInitializer(
+        HttpRequestInitializer httpRequestInitializer) {
+      this.httpRequestInitializer = httpRequestInitializer;
+      return this;
+    }
+
+    public GCSManualClientImpl build() throws IOException {
+      return new GCSManualClientImpl(this);
```

Comment:
Usually the builder will create the instance, rather than pasisng the builder to the constructor

Reply:
As discussed offline. This Nested class setup seems fine.
Only caveat being the `Impl` class need to public for integrations test.
There are two options via which we can make this `package-protected`
1. extract out the builder from the `Impl` class 
    * Have to make `Impl` public.
2. Created another wrapper Builder
    * Will have somany classees tied to each other but will serve the purpose.
    
For now I am annotating this class with `@VisibleForTesting`. Will delegate this decision to some later point in time.